citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -379,4 +384,14 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
         }
         return result;
     }
-}
+
+    @Override
+    public void deleteCitations(List<String> citationList) throws CannotFindCitationException {
```

Comment:
so, if one citation isn't found, the rest is not getting deleted? It should delete all other citations even if one can't be found.